diff --git a/base_partitions/flash.sh b/base_partitions/flash.sh
index dc9f2f2..27d3963 100755
--- a/base_partitions/flash.sh
+++ b/base_partitions/flash.sh
@@ -54,7 +54,7 @@ boot_path=$(read_path "Drag the boot image: ")
 fastboot flash boot "$boot_path" || { echo "Error flashing boot"; exit 1; }
 
 system_path=$(read_path "Drag the system image: ")
-fastboot flash rootfs "$system_path" || { echo "Error flashing rootfs"; exit 1; }
+fastboot flash rootfs "$system_path" || { echo "Error flashing system"; exit 1; }
 
 echo "Rebooting to EDL mode..."
 fastboot oem reboot-edl || { echo "Error rebooting to EDL"; exit 1; }
diff --git a/base_partitions/gpt_both0.bin b/base_partitions/gpt_both0.bin
index bbd6a4e..1ce01b2 100644
Binary files a/base_partitions/gpt_both0.bin and b/base_partitions/gpt_both0.bin differ
diff --git a/diffconfig b/diffconfig
index d232c1c..d493570 100644
--- a/diffconfig
+++ b/diffconfig
@@ -90,6 +90,6 @@ CONFIG_PACKAGE_uhttpd=y
 CONFIG_PACKAGE_uhttpd-mod-ubus=y
 CONFIG_PACKAGE_wireguard-tools=y
 # CONFIG_PER_FEED_REPO is not set
-# CONFIG_TARGET_IMAGES_GZIP is not set
-# CONFIG_TARGET_ROOTFS_SQUASHFS is not set
+# CONFIG_TARGET_ROOTFS_EXT4FS is not set
+CONFIG_TARGET_ROOTFS_PARTSIZE=128
 # CONFIG_PACKAGE_kmod-crypto-kpp is not set
diff --git a/msm89xx/image/Makefile b/msm89xx/image/Makefile
index 136ec84..42130ee 100644
--- a/msm89xx/image/Makefile
+++ b/msm89xx/image/Makefile
@@ -3,7 +3,6 @@
 include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/image.mk
 
-
 define Build/Compile
 	$(CP) $(LINUX_DIR)/COPYING $(KDIR)/COPYING.linux
 endef
@@ -18,21 +17,16 @@ define Build/aboot-img
 	mv $@.new $@
 endef
 
-define Build/sparse-img
-	img2simg  $@ $@.new
-	mv $@.new $@
-endef
-
 ### Devices ###
 define Device/Default
 	PROFILES := Default
 	KERNEL = kernel-bin | gzip | append-dtb
 	IMAGES := boot.img system.img
 	IMAGE/boot.img := append-kernel | aboot-img | append-metadata
-	IMAGE/system.img := append-rootfs | sparse-img | append-metadata
+	IMAGE/system.img := append-rootfs | append-metadata
 	DEVICE_DTS_DIR := $(DTS_DIR)/qcom
 	DEVICE_DTS = $$(SOC)-$(lastword $(subst _, ,$(1)))
-	CMDLINE := "earlycon console=ttyMSM0,115200 rw"
+	CMDLINE := "earlycon console=ttyMSM0,115200 root=PARTLABEL=rootfs ro rootwait"
 endef
 
 include $(SUBTARGET).mk
diff --git a/msm89xx/image/msm8916.mk b/msm89xx/image/msm8916.mk
index d2cf66d..b12b48c 100644
--- a/msm89xx/image/msm8916.mk
+++ b/msm89xx/image/msm8916.mk
@@ -3,8 +3,9 @@
 ifeq ($(SUBTARGET),msm8916)
 
 define Device/msm8916
-	SOC := msm8916
-	CMDLINE := "earlycon console=tty0 console=ttyMSM0,115200 root=/dev/mmcblk0p14 rw rootwait"
+  SOC := msm8916
+  # Usar PARTLABEL y root RO para squashfs:
+  CMDLINE := "earlycon console=tty0 console=ttyMSM0,115200 root=PARTLABEL=rootfs ro rootwait"
 endef
 
 define Device/yiming-uz801v3
diff --git a/packages/uz801-tweaks/Makefile b/packages/uz801-tweaks/Makefile
index 2e88ce7..6422194 100644
--- a/packages/uz801-tweaks/Makefile
+++ b/packages/uz801-tweaks/Makefile
@@ -17,7 +17,7 @@ define Package/$(PKG_NAME)
 	SECTION:=Utilities
 	CATEGORY:=Utilities
 	TITLE:=Defaults Setting For uz801v3
-	DEPENDS:=+rootfs-resizer +PACKAGE_luci:luci-proto-modemmanager +kmod-usb-lib-composite +rmtfs
+	DEPENDS:=+PACKAGE_luci:luci-proto-modemmanager +kmod-usb-lib-composite +rmtfs
 endef
 
 define Build/Configure
